app/Http/Controllers/admin/AdminController.php

```diff
@@ -310,25 +311,20 @@ public function export_results(Request $request)
         // @TODO: see if there is a better way to pass this from the form
         $filename = array_search('', $request->toArray());
         $export_function = $filename.'_query';
+        info('CSV downloads - ' . $filename . ' download requested');
 
         // Create an export object to run the query on
         $export = new Export();
         $query_result = $export->$export_function();
+        info('CSV downloads - '  . $filename . ' ' . count($query_result) . ' results found');
 
-        // Process the results of the query
-        $output = '';
-        foreach ($query_result as $row) {
-            $output .= implode(',', array_values(get_object_vars($row)))."\n";
-        }
-
-        // Build the csv
-        $filename = $filename.'-'.time().'.csv';
-        $headers = [
-          'Content-Type'        => 'text/csv',
-          'Content-Disposition' => 'attachment; filename="'.$filename.'"',
-        ];
+        // We need the result to be an array of an arrays, not an array of objects
+        $query_result = json_decode(json_encode($query_result), true);
```

Comment:
This is fine by me! The only downside here is if the total results were too large to keep in memory, in which case we'd want to have a traversable object that we could pass along and it could load and write piece-by-piece.

Reply:
@DFurnes I think that it definitely could be too large (30,000 rows?). Is the only alternative to iterate through each result, cast it as an array, and use `insertOne`?

Reply:
Yeah, you'd want to [chunk or use a cursor](https://laravel.com/docs/5.5/queries#chunking-results), and then you could `insertOne` one by one. The library also supports passing a [Traversable](http://php.net/manual/en/class.traversable.php), but not 100% sure that'd work out of the box.

Reply:
@DFurnes seems I can't chunk when using `DB::select` 😞  Would using a cursor work if `$query_result` is an array generated by something like this:
```
        $results = DB::select('SELECT u.id, u.first_name, u.last_name, u.email
                           FROM applications a
                           INNER JOIN users u on u.id = a.user_id
                           WHERE a.submitted is null');

        return $results;
```

Reply:
Hmm, I'd think `DB::select()` would just return a query instance that you could then chunk...  if you're running into trouble, let's try merging and deploying this & we can do further refactoring if we still run into issues.

Reply:
@DFurnes yeah with this:
```
    public function export_results(Request $request)
    {
        // Get the name of the function that runs the selected query
        // @TODO: see if there is a better way to pass this from the form
        $filename = array_search('', $request->toArray());
        $export_function = $filename.'_query';
        info('CSV downloads - ' . $filename . ' download requested');

        // Create an export object to run the query on
        $export = new Export();
        $query_result = $export->$export_function();
        info('CSV downloads - '  . $filename . ' ' . count($query_result) . ' results found');

        // Create the CSV file
        $writer = Writer::createFromPath($filename . '.csv', 'w+');
        $query_result->chunk(100, function ($results) {
            foreach ($results as $result) {
                dd($result);
            }
        });

        // Download the CSV file
        $writer->output($filename);
    }
```
I get:
`Call to a member function chunk() on array`